src/groupByKey.ts

```diff
@@ -2,6 +2,22 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+type Group = {
+  [key: string]: number | string;
+};
+
+export function groupByKey(items: Group[], key: string): GroupsMap<Group> {
```

Comment:
let's use more TS features

use generic type in function
change types for function arguments (you don't need your type `Group`, use `keyof` for `key`)

```suggestion
export function groupByKey<T>(items: Group[], key: string): GroupsMap<Group> {
```